### PR TITLE
base class BlobHandle

### PR DESCRIPTION
**Pull request by jckhang**
- JSONBlobHandle, CSVBlobHandle inherits from BlobHandle. Now we can use most function from BlobHandle

- Add api for download/upload blob

### CODE REVIEW

File: accern_xyme/accern_xyme.py [hunk 1]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+            raise ValueError(f"URI must be full: {self}")
+        if self.is_empty():
+            return None
+        fin, ctype = self._client._raw_request_bytes(METHOD_POST, "/uri", {
+            "uri": self._uri,
+            "pipeline": self.get_pipeline().get_id(),
+        })
+        return interpret_ctype(fin, ctype)
+
+    def list_files(self) -> List:
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1339]: List?

File: accern_xyme/accern_xyme.py [hunk 2]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+        if self.is_full():
+            raise ValueError(f"URI must not be full: {self}")
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1341]: return the file in a list?

File: accern_xyme/accern_xyme.py [hunk 3]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+        resp = self._client._request_json(
+            METHOD_GET, "/blob_files", {
+                "blob": self._uri,
+                "pipeline": self.get_pipeline().get_id(),
+            }, capture_err=False)
+        return resp["files"]
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1347]: response should be full URI blob handles
(response from the server should be URIs)

File: accern_xyme/accern_xyme.py [hunk 4]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+
+    def as_str(self) -> str:
+        return f"{self.get_uri()}"
+
+    def download_files(self, save_path: str) -> None:
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1352]: download_zip

File: accern_xyme/accern_xyme.py [hunk 5]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+            raise ValueError(f"URI must not be full: {self}")
+        cur_res, ctype = self._client._raw_request_bytes(
+            METHOD_GET, "/download_blob", {
+                "blob": self._uri,
+                "pipeline": self.get_pipeline().get_id(),
+            })
+        with open(save_path, "wb") as file_download:
+            file_download.write(cur_res.read())
+
+    def upload_file(self, from_path: str) -> List:
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1363]: upload_zip

File: accern_xyme/accern_xyme.py [hunk 6]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+        return resp["files"]
+
+    def as_str(self) -> str:
+        return f"{self.get_uri()}"
+
+    def download_files(self, save_path: str) -> None:
+        if self.is_full():
+            raise ValueError(f"URI must not be full: {self}")
+        cur_res, ctype = self._client._raw_request_bytes(
+            METHOD_GET, "/download_blob", {
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1356]: /download_zip

File: accern_xyme/accern_xyme.py [hunk 7]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+        with open(save_path, "wb") as file_download:
+            file_download.write(cur_res.read())
+
+    def upload_file(self, from_path: str) -> List:
+
+        with open(from_path, "rb") as fin:
+            zip_stream = io.BytesIO(fin.read())
+
+        resp = self._client._request_json(
+            METHOD_FILE, "/upload_blob", {
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1369]: /upload_zip

File: accern_xyme/accern_xyme.py [hunk 8]
```diff
@@ -1298,7 +1298,103 @@ def __repr__(self) -> str:
...
+                "blob": self._uri,
+                "pipeline": self.get_pipeline().get_id(),
+            }, files={
+                "file": zip_stream,
+            }, capture_err=False)
+        return resp["files"]
```
Comment (JosuaKrause) [inline accern_xyme/accern_xyme.py:1375]: if you return the files return blob handles

File: accern_xyme/accern_xyme.py (hunk 2 above)
Comment (jckhang) [reply accern_xyme/accern_xyme.py:1341]: if it's a full blob, the function is useless

**Pull request merged**